lib/ansible/modules/windows/win_updates.ps1

```diff
@@ -184,11 +187,30 @@ $update_script_block = {
                 Write-DebugLog -msg "Skipping hidden update $($update_info.title)"
                 continue
             }
-        
-            Write-DebugLog -msg "Adding update $($update_info.id) - $($update_info.title)"
-            $updates_to_install.Add($update) > $null
-        
-            $result.updates[$update_info.id] = $update_info
+
+            # Treat post-categories as matched if there are no post-categories
+            # to check
+            $post_cat_matched = ($post_categories.Count -eq 0)
+
+            ForEach ($update_cat in $update.Categories) {
+                $update_info.categories.Add($update_cat.Name) | Out-Null
```

Comment:
This should move out of this loop, if a match is found then only the categories that have been iterated would be returned. We should also not do these checks if `$post_cat_method` is initially `$true`.

Reply:
The loop on line 195 has been added to display the categories of an update, independent of whether there is post-category matching. This will return (especially during check/search mode) the list of categories for this update.

You asked about an easy way to see which categories are assigned to an update, this loop is providing this information. The inner loop does the checking.

Reply:
Of course (forehead slaps). I can do the outer loop as a one liner to stash the categories, but not so sure about the inner loop since it is a sub-string match. I'm more than happy for suggested code here.

Reply:
The main loop you have here is fine, you just need to get the categories outside this loop.

Reply:
See what you think of the changes.